fastapi/routing.py

```diff
@@ -356,7 +359,20 @@ def __init__(
         self.path = path
         self.endpoint = endpoint
         if isinstance(response_model, DefaultPlaceholder):
-            response_model = get_typed_return_annotation(endpoint)
+            return_annotation = get_typed_return_annotation(endpoint)
+            if get_origin(return_annotation) is Union:
+                union_args = get_args(return_annotation)
+                args = tuple(arg for arg in union_args if not issubclass(arg, Response))
+                if len(args) > 1:
+                    response_model = Union[args]
+                elif len(args) == 1:
+                    response_model = args[0]
+                else:
+                    response_model = None
+            elif issubclass(return_annotation, Response):
+                response_model = None
+            else:
+                response_model = return_annotation
```

Comment:
Those are the cases we talked about:
1. When there's a single `Response` class `response_model` becomes `None`.
2. If it's an `Union`, we filter the `Response` classes. (I'm not sure if this is what we want ???)

Reply:
Yeah, this gets quite complicated. Also because a Union could contain internal Unions, etc.

I think just the `lenient_issubclass(return_annotation, Response)`, to cover the simplest use cases.

And then I'll add a bit to the docs explaining that if the user wants to return and annotate something different than a Pydantic field, it needs a `response_model=None`.